lib/worker.js

```diff
@@ -150,6 +155,34 @@ async function updateDependency(upgrade) {
       newVersion);
   }
 
+  function getYarnLockFileName(packageFileName) {
+    const packagePath = packageFileName.split('/');
+    // Remove 'package.json'
+    const basePath = packagePath.slice(0, -1);
+    basePath.push('yarn.lock');
+    return basePath.join('/');
+  }
+
+  async function generateLockFile(newPackageJson) {
+    logger.debug('Generating new yarn.lock file');
+    const tmpDir = tmp.dirSync({ unsafeCleanup: true });
+    let yarnLock;
+    try {
+      fs.writeFileSync(path.join(tmpDir.name, 'package.json'), newPackageJson);
+      const npmrcContent = await api.getFileContent('.npmrc');
```

Comment:
Yarn also supports a project-specific `.yarnrc` file - https://github.com/yarnpkg/yarn/blob/master/src/registries/yarn-registry.js#L80

The locations where the `.yarnrc` file can be found are codified here - https://github.com/yarnpkg/yarn/blob/master/src/registries/npm-registry.js#L99

They are the same as the `.npmrc` file.